source/common/http/async_client_impl.cc

```diff
@@ -256,7 +256,10 @@ AsyncRequestImpl::AsyncRequestImpl(RequestMessagePtr&& request, AsyncClientImpl&
   } else {
     child_span_ = std::make_unique<Tracing::NullSpan>();
   }
-  child_span_->setSampled(options.sampled_);
+  if (options.sampled_) {
+    // Force span to get sampled; otherwise keep sampling status of the parent
+    child_span_->setSampled(true);
+  }
```

Comment:
I am sure if this is a good choice. Because this will prevent someone from using request options to disable sampling.

Reply:
I could add another field in options, `override_parent_sampling` which defaults to `true`, to keep the current behavior.

```
if (options.override_parent_sampling) {
	child_span_->setSampled(options.sampled_);
}
```

Reply:
🤔 sounds good to me.

Reply:
Would it be more simple make the `sampled_` optional: `absl::optional<bool> sampled_`?

The default value of `sampled_` still be `true` and we can set it to `absl::nullopt` in which case the child span will keep sampling status of parent span.

.

Reply:
Oh that's even better! I'll make the changes 😄